components/properties/src/provider.rs

```diff
@@ -139,76 +141,114 @@ pub mod key {
     );
 }
 
-//
-// UnicodeProperty
-//
-
-/// A set of characters with a particular property.
-#[icu_provider::data_struct(UnicodePropertyV1Marker)]
-#[derive(Debug, Eq, PartialEq, Clone)]
+/// Internal data type for [`CodePointSetData`]
+#[derive(Debug, Eq, PartialEq, Clone, yoke::Yokeable, zerofrom::ZeroFrom)]
 #[cfg_attr(feature = "datagen", derive(serde::Serialize))]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize))]
-pub struct UnicodePropertyV1<'data> {
+pub struct CodePointSet<'data> {
     /// The set of characters, represented as an inversion list
     #[cfg_attr(feature = "serde", serde(borrow))]
     pub inv_list: UnicodeSet<'data>,
```

Comment:
If/when you introduce a trait `CodePointSetLike` to represent the behavior of the CodePointSet data struct, would you be able to use that here somehow in the type of the field?  Both the field name and its current type suggest a singular concrete implementation type.